rotkehlchen/history/events.py

```diff
@@ -91,13 +92,19 @@ def _query_services_for_trades(self, filter_query: TradesFilterQuery) -> None:
         location = filter_query.location
         from_ts = filter_query.from_ts
         to_ts = filter_query.to_ts
+        excluded_instances: dict[Location, set[str]] = defaultdict(set[str])
 
         with self.db.conn.read_ctx() as cursor:
-            excluded_locations = {exchange.location for exchange in self.db.get_settings(cursor).non_syncing_exchanges}  # noqa: E501
+            for exchange_id in self.db.get_settings(cursor).non_syncing_exchanges:
+                excluded_instances[exchange_id.location].add(exchange_id.name)
 
         if location is not None:
-            if location not in excluded_locations:
-                self.query_location_latest_trades(location=location, from_ts=from_ts, to_ts=to_ts)
+            self.query_location_latest_trades(
+                location=location,
+                excluded_instances=excluded_instances.get(location, set()),
```

Comment:
Since is a defaultdict you can do 

```suggestion
                excluded_instances=excluded_instances[location],
```